lambda/src/main/scala/libretto/lambda/Shuffle.scala

```diff
@@ -298,7 +298,7 @@ class Shuffle[|*|[_, _]](using inj: BiInjective[|*|]) {
 
     def decompose1[X1, X2, Z](f: (X1 |*| X2) ~⚬ Z): Decomposition1[X1, X2, ?, ?, ?, ?, Z] =
       f match {
-        case Id()               => Decomposition1(Id(), Id(), TransferOpt.None(), implicitly)
+        case Id()               => Decomposition1(Id(), Id(), TransferOpt.None(), summon)
         case Bimap(Par(f1, f2)) => Decomposition1(f1, f2, TransferOpt.None(), implicitly)
         case Xfer(f1, f2, xfer) => Decomposition1(f1, f2, xfer, implicitly)
```

Comment:
I could not change the other two occurrences of `implicitly`. This results in very incomprehensible error messages. No clue what's going on here!

Reply:
I will have a look later.

Reply:
This works:

```scala
      f match {
        case Id()               => Decomposition1.make(Id(), Id(), TransferOpt.None())
        case Bimap(Par(f1, f2)) => Decomposition1.make(f1, f2, TransferOpt.None())
        case Xfer(f1, f2, xfer) => Decomposition1.make(f1, f2, xfer)
      }
```

where I also added

```scala
    object Decomposition1 {
      def make[X1, X2, Y1, Y2, Z1, Z2](
        f1: X1 ~⚬ Y1,
        f2: X2 ~⚬ Y2,
        g: TransferOpt[Y1, Y2, Z1, Z2],
      ): Decomposition1[X1, X2, Y1, Y2, Z1, Z2, Z1 |*| Z2] =
        Decomposition1(f1, f2, g, summon)
    }
```